### PR TITLE
Xenobiology Control Console

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
A free look/advanced camera console that can pick up and place slimes remotely, as well as deploy monkeys from stored monkey cubes. It can also pick up dead monkeys to recycle them directly into the machine. One of the biggest complaints about xenobio is how tedious and clunky it is dragging all the monkeys and slimes about, so hopefully this makes it more enjoyable.

The eye can only move inside the Xenobiology Lab so you can't use it to hunt badguys or dump slimes everywhere. However renaming areas to Xenobiology Lab will allow you to access them if you want to steal the blueprints and expand your territory.

DONE:
-Basically the whole machine. All the functions of it work in testing.
-Restricted to xenobiology only
-Icons
-Cerulean slimes now have a new reaction which gives you one use blueprints, for allowing you to expand your territory.

TODO:

-Changelog
-Put it on the map(s)
-Clean up some copypaste in the CameraOff part

Thanks to @AnturK for making these advanced camera machines in the first place

### CODE REVIEW

**Comment by Giacom** [comment]
Pro tip you can do

>git pull --rebase

to pull without getting the merge messages.

**Comment by Incoming5643** [comment]
Needs better sprites. Using abductor shuttle stuff on station just feels wrong.

**Comment by sirnat** [comment]
I think ill play science/xeno bio alot more now because of this.

**Comment by Bluespace** [comment]
Yeah this is absolutely amazing for xenobiology and removes some fuckwit tedium it had.

**Comment by Bluespace** [comment]
A "kill all slimed" button would be amazing, as it is right now you have to steal the RCD to make space tiles in xeno, or risk getting fucked hard when you manually kill them. A "deposit all slimes in processor" button would also eliminate a lot of boring clicking that achieves nothing but slowdown. I love this machine and think it'll totally fix xenobiology.

**Comment by KorPhaeron** [comment]
I was planning on redoing one of the cells into a little freeze chamber or space box when I actually map them in.

Speaking of tedium I built the monkey recyclers functions into the console

**Comment by Bluespace** [comment]
Eh, grey slime extracts remain incredibly superior for monkies anyway, shame I have to juggle stuff to inject 10 extracts and then click 30 times to load the camera, damn clickfest, really is.

**Comment by KorPhaeron** [comment]
The secret of monkey recycling is that it was never meant to actually be very efficient or useful, but to trick players into cleaning up the mob list by deleting the dead ones.

And yeah xenobio is still gonna have a ton of clicking unfortunatley, but way less dragging and godamn windoors

**Comment by KorPhaeron** [comment]
@RemieRichards @Razharas @Miauw62 @Jordie0608 @Cheridan 

This is ready to be reviewed

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 1]
```diff
@@ -0,0 +1,163 @@
+//Xenobio control console
+/mob/camera/aiEye/remote/xenobio
+	visible_icon = 1
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "camera_target"
+
+
+/mob/camera/aiEye/remote/xenobio/setLoc(var/t)
+	var/area/new_area = get_area(t)
+	if(new_area.name && new_area.name == "Xenobiology Lab")
```
Comment (Razharas) [inline code/modules/research/xenobiology/xenobio_camera.dm]: Dont areas have type?
I was sure all our areas are different types

**Comment by Bluespace** [comment]
m e r g e  i t

File: code/modules/research/xenobiology/xenobio_camera.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/research/xenobiology/xenobio_camera.dm]: Yeah but doing it by name allows for territory expansion via blueprint use so you can turn the game into dungeon keeper/zombie master and drop slimes on everyone until they come murder you

Comment (Jordie0608) [inline code/modules/research/xenobiology/xenobio_camera.dm]: What if someone renames the xeno lab then?

Comment (KorPhaeron) [inline code/modules/research/xenobiology/xenobio_camera.dm]: Then the console is unusable until you fix it.

I guess I could add an or to check for the Xenobio area type so it's always usable there no matter what.

**Comment by Razharas** [comment]
Eh whatever

**Comment by KorPhaeron** [comment]
Fuck I was about to fix the thing you told me to I am aliterally committing it

**Comment by GunHog** [comment]
Question, is it possible to replace these consoles if lost?

**Comment by KorPhaeron** [comment]
Nope

**Pull request merged**